### PR TITLE
feat(log): add logging library using `spdlog`

### PR DESCRIPTION
**Pull request by Miou-zora**
I choose [`spdlog`](https://github.com/gabime/spdlog) for a logging library because it's the most common logging lib used for game engine and it looks perfect for our needs.

So I've swited all std::cout and std::cerr of this project for logging functions that you can see in `Logger.hpp`.

I also added a new utility namespace for strings.

All game engine I know that use `spdlog`: [SupernovaEngine](https://github.com/skaarj1989/SupernovaEngine), [RendererEngine](https://github.com/JeanPhilippeKernel/RendererEngine), [ElvenEngine](https://github.com/denyskryvytskyi/ElvenEngine), [ezEngine](https://github.com/ezEngine/ezEngine), other game engine uses customs logger or don't provide infos about it.

### CODE REVIEW

File: src/plugin/log/src/Logger.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "spdlog/spdlog.h"
+
+namespace ES::Plugin::Log {
```
Comment (ripel2) [inline src/plugin/log/src/Logger.hpp:5]: logger is a plugin??

File: src/plugin/vk-wrapper/src/VkWrapper.cpp [hunk 2]
```diff
@@ -40,15 +42,15 @@ void VkWrapper::PrintAvailableExtensions()
     std::vector<VkExtensionProperties> extensions(extensionCount);
     vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, extensions.data());
 
-    std::cout << "available extensions (" << extensionCount << "):" << std::endl;
```
Comment (MasterLaplace) [inline src/plugin/vk-wrapper/src/VkWrapper.cpp:43]: Perhaps you could concatenate the whole list into one message and print it out ?

File: src/plugin/vk-wrapper/src/debugMessenger/DebugMessenger.cpp [hunk 3]
```diff
@@ -25,8 +27,16 @@ VKAPI_ATTR VkBool32 VKAPI_CALL DebugMessenger::Callback(VkDebugUtilsMessageSever
                                                         const VkDebugUtilsMessengerCallbackDataEXT *pCallbackData,
                                                         void *pUserData)
 {
-    std::cerr << "validation layer: " << pCallbackData->pMessage << std::endl;
-
+    std::unordered_map<VkDebugUtilsMessageSeverityFlagBitsEXT, ES::Plugin::Log::Level> severityMap = {
```
Comment (MasterLaplace) [inline src/plugin/vk-wrapper/src/debugMessenger/DebugMessenger.cpp:30]: Could this map be placed as a constant in the `DebugMessenger.hpp` header or out of function, above ?

File: src/plugin/utils/tests/StringTest.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#include <gtest/gtest.h>
+
+#include "String.hpp"
+
+using namespace ES::Plugin::Utils;
+
+TEST(String, Split)
+{
+    { // One split
```
Comment (MasterLaplace) [inline src/plugin/utils/tests/StringTest.cpp:9]: It count as multiple tests or just one ?

File: src/plugin/utils/src/String.hpp [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace ES::Plugin::Utils::String {
+std::vector<std::string> Split(const std::string &str, const char &delimiter)
```
Comment (MasterLaplace) [inline src/plugin/utils/src/String.hpp:6]: No CPP file for utils ?

File: src/plugin/scene/tests/SceneTest.cpp [hunk 6]
```diff
@@ -5,6 +5,8 @@
 
 #include "Engine.hpp"
 
+#include "String.hpp"
+
 #include "Scene.hpp"
 
 using namespace ES::Plugin::Scene;
```
Comment (MasterLaplace) [inline src/plugin/scene/tests/SceneTest.cpp:12]: Still used in the test

File: src/plugin/log/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+.xmake/*
+build/
```
Comment (MasterLaplace) [inline src/plugin/log/.gitignore:2]: Where's the EOF, did you eat it?

File: src/plugin/log/src/Logger.cpp [hunk 8]
```diff
@@ -0,0 +1 @@
+#include "Logger.hpp"
```
Comment (MasterLaplace) [inline src/plugin/log/src/Logger.cpp:1]: useful ?

File: src/plugin/log/src/Logger.hpp (hunk 1 above)
Comment (Miou-zora) [reply src/plugin/log/src/Logger.hpp:5]: It's more like utils function you're right

Comment (Miou-zora) [reply src/plugin/log/src/Logger.hpp]: Done

File: src/plugin/vk-wrapper/src/VkWrapper.cpp (hunk 2 above)
Comment (Miou-zora) [reply src/plugin/vk-wrapper/src/VkWrapper.cpp:43]: Done

File: src/plugin/vk-wrapper/src/debugMessenger/DebugMessenger.cpp (hunk 3 above)
Comment (Miou-zora) [reply src/plugin/vk-wrapper/src/debugMessenger/DebugMessenger.cpp]: Done

File: src/plugin/utils/tests/StringTest.cpp (hunk 4 above)
Comment (Miou-zora) [reply src/plugin/utils/tests/StringTest.cpp:9]: Multiple tests, I'll split it into multiple `TEST` call

Comment (MasterLaplace) [reply src/plugin/utils/tests/StringTest.cpp:9]: Good to know thanks

Comment (Miou-zora) [reply src/plugin/utils/tests/StringTest.cpp]: Done

File: src/plugin/utils/src/String.hpp (hunk 5 above)
Comment (Miou-zora) [reply src/plugin/utils/src/String.hpp]: Oh I forgor

File: src/plugin/scene/tests/SceneTest.cpp (hunk 6 above)
Comment (Miou-zora) [reply src/plugin/scene/tests/SceneTest.cpp:12]: What do you mean ?

**Review by MasterLaplace — APPROVED** [review]

**Review by ripel2 — APPROVED** [review]

**Pull request merged**